modules/editorial-metadata/editorial-metadata.php

```diff
@@ -46,10 +46,36 @@ public function init() {
 		// Register the taxonomy we use for Editorial Metadata with WordPress core
 		$this->register_taxonomy();
 
+		EditEditorialMetadata::init();
+
 		// Load CSS and JS resources that we probably need in the admin page
 		add_action( 'admin_enqueue_scripts', [ $this, 'action_admin_enqueue_scripts' ] );
 
-		EditEditorialMetadata::init();
+		// Load block editor JS
+		add_action( 'enqueue_block_editor_assets', [ $this, 'load_scripts_for_block_editor' ], 9 /* Load before custom status module */ );
+
+		// Load block editor CSS
+		add_action( 'enqueue_block_editor_assets', [ $this, 'load_styles_for_block_editor' ] );
+
+		// Register the post meta for each editorial metadata term
+		add_action( 'init', [ $this, 'register_editorial_metadata_terms_as_post_meta' ] );
+	}
+
+	/**
+	 * Register the post meta for each editorial metadata term
+	 */
+	public function register_editorial_metadata_terms_as_post_meta() {
+		$editorial_metadata_terms = $this->get_editorial_metadata_terms();
+
+		foreach ( $editorial_metadata_terms as $term ) {
+			$post_meta_key = $this->get_postmeta_key( $term );
+			$post_meta_args = $this->get_postmeta_args( $term );
+
+			// ToDo: Fix the bug where if a post type is later unsupported, the editorial metadata fields still show up for that post type
```

Comment:
Can you explain this issue? I'd think if a post type was unsupported, it wouldn't show up in `get_supported_post_types()` and then wouldn't get registered.

Reply:
Due to the way the `register_post_meta` operation works, if a post meta has been registered for say a `page` and then later support for a `page` is removed then that post_meta would still show up for a page. I'm leaning towards checking the supported post type in the block editor to manually stop the editorial metadata from loading.